src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -1276,5 +1303,57 @@ private bool TryTranslateSumResultOperator(PipelineExpression node, out BsonValu
             result = null;
             return false;
         }
+
+        private static class TemplateHelper
+        {
+            private const string MapTemplateStartDelimiter = "#{{";
+            private const string MapTemplateEndDelimiter = "}}#";
+            private const string MapTemplate = MapTemplateStartDelimiter + "{{{0}}}" + MapTemplateEndDelimiter;
+
+            private static string GetNotReplacedTemplate(string value)
+            {
+                var startIndexOf = value.IndexOf(MapTemplateStartDelimiter, StringComparison.Ordinal);
+                var endIndexOf = value.IndexOf(MapTemplateEndDelimiter, StringComparison.Ordinal);
+                if (startIndexOf != -1 && endIndexOf != -1 && endIndexOf > startIndexOf)
+                {
+                    endIndexOf = endIndexOf + MapTemplateEndDelimiter.Length;
+                    if (endIndexOf <= value.Length) return value.Substring(startIndexOf, endIndexOf - startIndexOf);
+                }
+
+                return null;
+            }
+
+            public static BsonValue ReplaceTemplates(BsonValue bsonValue, Dictionary<string,string> mapHistory)
+            {
+                var value = bsonValue.ToString();
+                foreach (var map in mapHistory)
+                {
+                    value = value.Replace(string.Format(MapTemplate, map.Value), map.Key);
```

Comment:
It's really hard to figure out what string.Format is doing here. Why even use string.Format at all? Why not just use something like:

    var oldValue = $"#<{map.Value}>#"; // assuming using "<" in templates instead of "{" which requires escaping
    var newValue = map.Key;
    value = value.Replace(oldValue, newValue);

For that matter, why doesn't map.Value already have the finished template already? Why do we have to do any formatting here?

Reply:
Agree, it can be replaced on your suggestion.